neural_compressor/onnxrt/algorithms/weight_only/rtn.py

```diff
@@ -0,0 +1,267 @@
+        scheme (str, optional): quantization scheme. Defaults to "asym".
+        dtype (str, optional): data type. Defaults to "int".
+        ratio (float, optional): percentile of clip. Defaults to 1.0.
+
+    Returns:
+        output: quant-dequant weight
+    """
+    org_shape = data.shape
+    weight, scale, zp = quant_tensor(data, num_bits, group_size, scheme, dtype, ratio)
+    return np.reshape(scale * (weight - zp), org_shape)
+
+
+def rtn_quantize(
+    model: Union[onnx.ModelProto, ONNXModel, Path, str],
+    weight_config={},
+    num_bits=4,
+    group_size=32,
+    scheme="asym",
+    ratios={},
+    accuracy_level=0,
+    providers=["CPUExecutionProvider"],
+):
+    """Quant the model with round to nearst method.
+
+    Args:
+        model (onnx.ModelProto or ONNXModel or ): onnx model
+        weight_config (dict): quantization config
+                For example,
+                weight_config = {
+                    '("/h.0/mlp/fc_out/MatMul", "MatMul")':
+                        {
+                            'bits': 4,
+                            'group_size': 32,
+                            'scheme': 'sym',
+                            'algorithm': 'RTN'
+                        }
+                }
+        num_bits (int, optional): num_bits. Default is 4.
+        group_size (int, optional): how many elements share one scale/zp. Default is 32.
+        scheme (str, optional): sym or asym. Defaults to "asym".
+        ratios (dict, optional): percentile of clip. Defaults to {}.
+        accuracy_level (int): accuracy level. Support 0 (unset), 1(fp32 compute type of jblas kernel),
+                              2 (fp16 compute type of jblas kernel), 3 (bf16 compute type of jblas kernel),
+                              4 (int8 compute type of jblas kernel)
+        providers (list): providers to use
+
+    Returns:
+        model: fake quantized ONNXModel
+    """
+    if not isinstance(model, ONNXModel):
+        model = ONNXModel(model)
+    base_dir = os.path.dirname(model.model_path) if model.model_path is not None else ""
+    new_nodes = []
+    remove_nodes = []
+    total_num = len([i for i in model.nodes() if i.op_type in ["MatMul"]])
+    curr_id = 0
+    for node in model.nodes():
+        if node.op_type in ["MatMul"]:
+            curr_id += 1
+            simple_progress_bar(total_num, curr_id)
+
+        # check op_type of node is MatMul
+        # check dim 1 of input is weight tensor
+        # check node config is RTNWeightQuantConfig
+        # check node weight_dtype config is not fp32
+        if (
+            node.op_type in ["MatMul"]  # check op_type of node is MatMul
+            and model.get_initializer(node.input[1]) is not None
+            and isinstance(weight_config.get((node.name, node.op_type), {}), RTNWeightQuantConfig)
+            and weight_config.get((node.name, node.op_type), {}).weight_dtype.lower() != "fp32"

```

Comment:
if weight_config is {}, '.weight_dtype' will raise error

Reply:
if weight_config is {}, `isinstance(weight_config.get((node.name, node.op_type), {}), RTNWeightQuantConfig)` will be False and L190 will not be executed